test/spl/degree_elevation_test.cc

```diff
@@ -0,0 +1,516 @@
+/* Copyright 2018 Chair for Computational Analysis of Technical Systems, RWTH Aachen University
+
+This file is part of SplineLib.
+
+SplineLib is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 3 of the License.
+
+SplineLib is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with SplineLib.  If not, see
+<http://www.gnu.org/licenses/>.
+*/
+
+#include "gmock/gmock.h"
+
+#include "b_spline.h"
+#include "nurbs.h"
+#include "random_b_spline_generator.h"
+#include "random_nurbs_generator.h"
+
+using testing::Test;
+
+class BSplineFig5_35 : public Test {  // NOLINT
+ public:
+  BSplineFig5_35() {
+    std::array<Degree, 1> degree = {Degree{3}};
+    KnotVectors<1> knot_vector_before = {std::make_shared<baf::KnotVector>(
+        baf::KnotVector({ParamCoord{0}, ParamCoord{0}, ParamCoord{0}, ParamCoord{0}, ParamCoord{0.3}, ParamCoord{0.6},
+                         ParamCoord{1}, ParamCoord{1}, ParamCoord{1}, ParamCoord{1}}))};
+    std::vector<baf::ControlPoint> control_points = {
+        baf::ControlPoint(std::vector<double>({1.0, 0.0})),
+        baf::ControlPoint(std::vector<double>({0.0, 2.0})),
+        baf::ControlPoint(std::vector<double>({1.0, 2.0})),
+        baf::ControlPoint(std::vector<double>({3.0, 2.0})),
+        baf::ControlPoint(std::vector<double>({4.0, 1.0})),
+        baf::ControlPoint(std::vector<double>({3.0, 0.0}))
+    };
+    original_ = std::make_shared<spl::BSpline<1>>(knot_vector_before, degree, control_points);
+    after_elevation_ = std::make_shared<spl::BSpline<1>>(*original_);
+    after_elevation_->ElevateDegree(0);
+  }
+
+ protected:
+  std::shared_ptr<spl::BSpline<1>> original_;
+  std::shared_ptr<spl::BSpline<1>> after_elevation_;
+};
+
+TEST_F(BSplineFig5_35, HasELevatedDegree) {  // NOLINT
+  ASSERT_THAT(after_elevation_->GetDegree(0).get(), original_->GetDegree(0).get() + 1);
+}
+
+TEST_F(BSplineFig5_35, HasMoreKnots) {  // NOLINT
```

Comment:
How many more? "Number of non-zero knot spans" more?

Reply:
It actually is `NumberNonZeroKnotSpans` = `NumberDifferentKnots` - 1.

So, is `HasNumberOfDifferentKnotsMoreKnots` okay, or should I apply the actual number?